src/sage/combinat/posets/linear_extensions.py

```diff
@@ -252,6 +252,73 @@ def is_greedy(self):
                         return False
         return True
 
+    def is_supergreedy(self):
+        r"""
+        Return ``True`` if the linear extension is supergreedy.
+        
+        A linear extension `[x_1<x_2<...<x_t]` of a finite ordered
+        set `P=(P, <)` is *super greedy* if it can be obtained using
+        the following procedure: Choose `x_1` to be a minimal
+        element of `P`; suppose `x_1,...,x_i` have been chosen;
+        define `p(x)` to be the largest `j\leq i` such that `x_j<x`
+        if such a `j` exists and 0 otherwise; choose `x_{i+1}`
+        to be a minimal element of `P-\{x_1,...,x_i\}` which
+        maximizes `p`.
+
+        Informally, a linear extension is supergreedy if it "always
+        goes up and receedes the least"; in other words, supergreedy
+        linear extensions are depth-first linear extensions.
+        For more details see [KTZ1987]_.
+
+        EXAMPLES::
+        
+            sage: X = [0,1,2,3,4,5,6]
+            sage: Y = [[0,5],[1,4],[1,5],[3,6],[4,3],[5,6],[6,2]]
+            sage: P = Poset((X,Y), cover_relations = True, facade=False)
+            sage: for l in P.linear_extensions():
+            ....:     if l.is_supergreedy():
+            ....:         print(l)
+            [1, 4, 3, 0, 5, 6, 2]
+            [0, 1, 4, 3, 5, 6, 2]
+            [0, 1, 5, 4, 3, 6, 2]
+
+            sage: Q = posets.PentagonPoset()
+            sage: for l in Q.linear_extensions():
+            ....:     if not l.is_supergreedy():
+            ....:         print(l)
+            [0, 2, 1, 3, 4]
+
+        TESTS::
+
+            sage: T = Poset()
+            sage: T.linear_extensions()[0].is_supergreedy()
+            True
+        """
+        P = self.poset()
+        H = P.hasse_diagram()
+
+        def next_elements(H, linext):
+            k = len(linext)
+            S = []
+            while not S:
+                if not k:
+                    S = [x for x in H.sources() if x not in linext]
+                else:
+                    S = [x for x in H.neighbor_out_iterator(linext[k-1]) if x not in linext and all(low in linext for low in H.neighbor_in_iterator(x))]
```

Comment:
such a long line should better be split. I wonder whether our linters ought to pick this up, @tobiasdiez ?

Reply:
It gives an error above 160 characters, see https://github.com/sagemath/sage/blob/fbb412787f6a762442c41a2d7cd01715d9994cb9/src/tox.ini#L130. There are quite a few instances of long lines already in the code, and it is also not clear to me what an optimal line length should be. For example "Black defaults to 88 characters per line, which happens to be 10% over 80. This number was found to produce significantly shorter files than sticking with 80 (the most popular)"